src/components/workspace/Editor.tsx

```diff
@@ -8,51 +9,140 @@ import 'brace/mode/javascript';
 import 'brace/theme/cobalt';
 
 import { LINKS } from '../../utils/constants';
+import {
+  ICodeDelta,
+  Input,
+  IPosition,
+  ISelectionRange,
+  KeyboardCommand
+} from '../sourcecast/sourcecastShape';
 import { checkSessionIdExists } from './collabEditing/helper';
 /**
  * @property editorValue - The string content of the react-ace editor
  * @property handleEditorChange  - A callback function
  *           for the react-ace editor's `onChange`
  * @property handleEvalEditor  - A callback function for evaluation
  *           of the editor's content, using `slang`
+ * @property editorReadonly - Used for sourcecast only
```

Comment:
I'm not sure how to go about splitting this up as I am using a lot of APIs from `AceEditor`.
Should I wrap `Editor` with something like `SourcecastEditor` and reach into `Editor` with Refs?

Reply:
Let's duplicate `Editor` and apply your changes into `SourcecastEditor`. I think it would be easier to ensure that the current `Editor` works, than to add code to it. We will work out the refactoring later.